### PR TITLE
Api 32142 error when missing active duty dates

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

- Updates 526.json schema and associated documentation to disallow null or empty values or missing fields for 526 form serviceInformation.servicePeriods activeDutyBeginDate & activeDutyEndDate, after it was discovered that the EVSS docker container requires these values.

## Related issue(s)
[API-32142](https://jira.devops.va.gov/browse/API-32142)

## Testing done

- Local testing using Postman to submit to our 526 endpoint.
- Rspec

## What areas of the site does it impact?
modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
modules/claims_api/config/schemas/v2/526.json
modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb
modules/claims_api/app/swagger/claims_api/v2/production/swagger.json

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json [hunk 1]
```diff
@@ -5806,7 +5806,7 @@
                       "status": "404",
                       "detail": "Resource not found",
                       "source": {
-                        "pointer": "/Users/rockwellrice/Documents/oddball/vets-api/modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:74:in `attachments'"
+                        "pointer": "/Users/matthewchristianson/development/vets-api/modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:74:in `attachments'"
                       }
```
[the remark this answers is not in the document]
Comment (mchristiansonVA) [reply modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json]: I just manually edited to remove the `source: pointer` section here for now.

File: modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json [hunk 2]
```diff
@@ -5804,10 +5804,7 @@
                     {
                       "title": "Resource not found",
                       "status": "404",
-                      "detail": "Resource not found",
-                      "source": {
-                        "pointer": "/Users/rockwellrice/Documents/oddball/vets-api/modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:74:in `attachments'"
-                      }
+                      "detail": "Resource not found"
                     }
```
Comment (rockwellwindsor-va) [inline modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:5808]: I think we still need the source and pointer value in there, just shortened down, perhap something like `/modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb:74:in ...`

Comment (mchristiansonVA) [reply modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:5808]: Apologies, I think I misinterpreted your original comment; added back and shortened to remove local directory reference in latest push.

Comment (rockwellwindsor-va) [reply modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json:5811]: no worries, I could have been clearer

**Review by rockwellwindsor-va — APPROVED** [review]
Tested w/ Postman, LGTM!

**Comment by FonzMP** [comment]
Needs production swagger updates - 526 is now live in production. @mchristiansonVA also for awareness @rockwellwindsor-va

**Review by rockwellwindsor-va — COMMENTED** [review]
@mchristiansonVA  There are some remaining validations in the `disability_compensation_validations.rb` file that conditionally are checking for these dates, since they were previously not required.  I think those could likely be cleaned out here (although I only looked over the related code for these two fields).  Specifically:

`activeDutyBeginDate`
1. disability_compensation_validations.rb:472 (conditional check)
2. disability_compensation_validations.rb:491 (before the || on the conditional)
3. disability_compensation_validations.rb:551 (conditional check)
4. disability_compensation_validations.rb:642 (conditional check)
5. disability_compensation_validations.rb:750 (conditional check)

`activeDutyEndDate`
1. disability_compensation_validations.rb:556 (conditional check)
2. disability_compensation_validations.rb:567 (before the  || on the conditional check)
3. disability_compensation_validations.rb:595 (conditional check)
4. disability_compensation_validations.rb:595 (before the && in the conditional check)

**Review by FonzMP — APPROVED** [review]
:shipit:

**Review by FonzMP — APPROVED** [review]
:shipit:

**Pull request merged**